Gemfile

```diff
@@ -147,6 +147,7 @@ def override_gem(name, *args)
     dependencies.delete(dependency)
 
     gem name, *args
+    Bundler::UI::Shell.new.warn "Caution, using override_gem for #{name} with args: #{args.inspect}, review Gemfile.dev.rb if this is not desired!"
```

Comment:
Ugh... why am I here on my day off... but this might get merged in a day... so...

As a possible compromise (and possibly introducing a bit of scope creep), maybe make as part of `bin/setup` we generate the Gemfile.dev.rb file that includes this warning in some other fashion (not sure how, possibly monkey patch the method again and call `super`), but by doing this, we then can:

* Makes this not show up in prod
* For snowflakes, like @kbrock, they can just delete the method
* For people who want a more verbose output, like @cben, they can then update the warning themselves

I will disagree with @bdunne 's comment a bit though:

> I think it would be good to move it up one line.

A)  Because I think it is harder to implement with a super (though, with the above suggestion, you could do whatever I guess)
B)  You only show this warning when it is successfully added, not when it fails.

Both might be subjective, so mostly I am just weighing in, but I think this suggestion (regardless of how it ends up being implemented), covers all of @bdunne 's concerns as well.

That said, this doesn't help folks who already have a Gemfile.dev.rb (and we wouldn't want to override that with this implementation), so there is that to consider with this option as well.